File: orbees-api/Models/Transaction.cs
using Api.Models.Common;
using Api.Models.Enums;

namespace Api.Models
{
    public class Transaction : AuditableEntity
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public required string Title { get; set; }

        public string? OriginalDescription { get; set; }

        public string? Description { get; set; }

        public decimal Amount { get; set; }
        public DateTime TransactionDate { get; set; }

        public TransactionType Type { get; set; }

        public TransactionOrigin Origin { get; set; }

        public string? MerchantDocument { get; set; }

        public bool IsActive { get; set; } = true;

        public Guid UserId { get; set; }
        public User User { get; set; } = null!;

        public Guid? BankAccountId { get; set; }
        public BankAccount? BankAccount { get; set; }

        public Guid? CategoryId { get; set; }
        public Category? Category { get; set; }

        public Guid? GroupCategoryId { get; set; }
        public Category? GroupCategory { get; set; }

        public Guid? GroupId { get; set; }
        public Group? Group { get; set; }

        public bool GroupLinkActive { get; set; } = true;
    }
}

File: orbees-frontend/src/interfaces/transaction.ts
import { TransactionType, TransactionOrigin } from "./enums";

export interface TransactionReadDto {
  id: string;
  title: string;
  originalDescription?: string;
  description?: string;
  amount: number;
  transactionDate: string;
  type: TransactionType;
  transactionOrigin: TransactionOrigin;
  merchantDocument?: string;
  categoryId?: string;
  categoryName?: string;
  categoryColor?: string;
  categoryIcon?: string;
  groupCategoryId?: string;
  groupCategoryName?: string;
  groupCategoryColor?: string;
  groupCategoryIcon?: string;
  groupId?: string;
  groupName?: string;
  memberName?: string;
  groupLinkActive: boolean;
  bankAccountId?: string;
  bankAccountName?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface TransactionCreateDto {
  title: string;
  description?: string;
  amount: number;
  transactionDate: string;
  type: TransactionType;
  merchantDocument?: string;
  bankAccountId?: string;
  categoryId?: string;
  groupId?: string;
  groupCategoryId?: string;
}

export interface TransactionUpdateDto {
  title?: string;
  description?: string;
  categoryId?: string;
  removeCategoryId?: boolean;
  groupCategoryId?: string;
  removeGroupCategoryId?: boolean;
  groupId?: string;
  removeGroupId?: boolean;
}

export interface TransactionPreviewDto {
  title: string;
  originalDescription?: string;
  amount: number;
  transactionDate: string;
  type: TransactionType;
  merchantDocument?: string;
  suggestedCategoryId?: string;
  suggestedCategoryName?: string;
  categoryId?: string;
  groupId?: string;
  groupCategoryId?: string;
}

export interface TransactionImportDto {
  bankAccountId?: string;
  transactions: TransactionPreviewDto[];
}

export interface TransactionBulkCreateDto {
  transactions: TransactionCreateDto[];
}

export interface PagedResultDto<T> {
  items: T[];
  currentPage: number;
  totalPages: number;
  totalItems: number;
  pageSize: number;
}
